### PR TITLE
Add check if partition/tile cached

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Add implementation to VersionLayerClientImpl
if partition/tile cached. Added check to
DataCacheRepository if data handle cached.
Add tests for PartitionCacheRepository to check
if partition is cached.

Relates-To: OLPEDGE-2111

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp [hunk 1]
```diff
@@ -279,6 +282,29 @@ bool PartitionsCacheRepository::ClearPartitionMetadata(
...
+        });
+
+    if (cached_partition.empty()) {
+      return false;
+    }
+    auto partition = boost::any_cast<model::Partition>(cached_partition);
+    auto data_key = CreateKey(hrn, layer_id, partition.GetDataHandle());
+    OLP_SDK_LOG_DEBUG_F(kLogTag, "IsPartitionCached data -> '%s'", key.c_str());
+    return cache_->Contains(data_key);
+  }
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp:304]: Looks misleading to me. This method should check is metadata cached only.
Also, I think, calling `cache_->Get()` after `cache_->Contains()` is an overhead.

Comment (andescu) [reply olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp:304]: We need to check if the data handle is also cached. Knowing that the metadata is cached does not bring anything.

File: olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp [hunk 2]
```diff
@@ -519,6 +519,18 @@ bool VersionedLayerClientImpl::RemoveFromCache(const geo::TileKey& tile) {
...
+  repository::PartitionsCacheRepository cache_repository(catalog_,
+                                                         settings_.cache);
+  return cache_repository.IsPartitionCached(catalog_version_.load(),
+                                            partition_id, layer_id_);
+}
+
+bool VersionedLayerClientImpl::IsCached(const geo::TileKey& tile) const {
+  auto partition_id = tile.ToHereTile();
+  return IsCached(partition_id);
+}
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp:532]: This is wrong. You need to use quad tree index for a tile, not a single partition entry from cache!

File: olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp [hunk 3]
```diff
@@ -519,6 +519,18 @@ bool VersionedLayerClientImpl::RemoveFromCache(const geo::TileKey& tile) {
   return RemoveFromCache(partition_id);
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp:519]: After our switch to quad tree index this is also wrong and will probably never delete anything. Please fix!

File: olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp (hunk 1 above)
Comment (kzvieriev) [reply olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp:304]: Yeah, but `PartitionsCacheRepository` isn't a best place to do it.
I would prefer to check data handle in `VersionedLayerClientImpl` instead.

File: olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp (hunk 3 above)
Comment (LiubovDidkivska) [reply olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp:519]: Will be fixed in separate PR

File: olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp (hunk 1 above)
Comment (LiubovDidkivska) [reply olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp]: moved

File: olp-cpp-sdk-dataservice-read/tests/DataCacheRepositoryTest.cpp [hunk 4]
```diff
@@ -66,6 +67,31 @@ TEST(PartitionsCacheRepositoryTest, DefaultExpiry) {
...
+    SCOPED_TRACE("Is not cached");
+
+    std::shared_ptr<cache::KeyValueCache> cache =
+        olp::client::OlpClientSettingsFactory::CreateDefaultCache({});
+    repository::DataCacheRepository repository(hrn, cache);
+
+    const auto result = repository.IsCached(layer, kDataHandle);
+
+    EXPECT_FALSE(result);
+  }
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/tests/DataCacheRepositoryTest.cpp:94]: these test cases are not related to DefaultExiry, lets separate them
